### PR TITLE
added support to Lighthouse for new backend

### PR DESCRIPTION
**Pull request by jroebu14**
## Hey, I just made a Pull Request!

Added support for the new backend system for the Lighthouse plugin.

Fixes https://github.com/backstage/backstage/issues/18655

Original issue: https://github.com/backstage/backstage/issues/18301

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: plugins/lighthouse-backend/README.md [hunk 1]
```diff
@@ -58,6 +58,23 @@ export default async function createPlugin(env: PluginEnvironment) {
...
+In your `packages/backend/src/index.ts` make the following changes:
+
+```diff
+  import { createBackend } from '@backstage/backend-defaults';
++ import { lighthousePlugin } from '@backstage/plugin-lighthouse-backend';
+  const backend = createBackend();
+  // ... other feature additions
++ backend.add(
++   lighthousePlugin(),
++ );
```
Comment (Rugvip) [inline plugins/lighthouse-backend/README.md:74]:
```suggestion
+ backend.add(lighthousePlugin());
```

Might as well right? 😁

File: plugins/lighthouse-backend/src/plugin.ts [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+      deps: {
+        config: coreServices.config,
+        discovery: coreServices.discovery,
+        logger: coreServices.logger,
+        scheduler: coreServices.scheduler,
+        tokenManager: coreServices.tokenManager,
+      },
+      async init({ config, discovery, logger, scheduler, tokenManager }) {
+        const winstonLogger = loggerToWinstonLogger(logger);
+        const catalogClient = new CatalogClient({ discoveryApi: discovery });
```
Comment (Rugvip) [inline plugins/lighthouse-backend/src/plugin.ts:43]: Let's depend on the `catalogServiceRef` from `'@backstage/plugin-catalog-node/alpha'` to access this instead.

**Review by Rugvip — CHANGES_REQUESTED** [review]
Thank you, looks good! 👍 

Just some minor nits

File: plugins/lighthouse-backend/README.md (hunk 1 above)
Comment (jroebu14) [reply plugins/lighthouse-backend/README.md:74]: much prettier 😄

**Review by Rugvip — APPROVED** [review]
Excellent, let's :shipit:!

**Pull request merged**